docs/source/core_services/drivers/driver_configuration/modbus-tk-driver.rst

```diff
@@ -0,0 +1,263 @@
+to streamline configuration file creation and maintenance.
+
+The Modbus-TK driver is mostly backward-compatible with the parameter definitions in the original
+Modbus driver's configuration (.config and .csv files).
+If the config file's parameter names use the Modbus driver's name conventions, they are
+translated to the Modbus-TK name conventions, e.g. a Modbus CSV file's "Point Address" is
+interpreted as a Modbus-TK "Address". Backward-compatibility exceptions are:
+
+    - If the config file has no **port**, the default is 0, not 502.
+    - If the config file has no **slave_id**, the default is 1, not 0.
+    - An **endian** value other than **big** (the default) will not be interpreted correctly.
+
+driver_config
+*************
+
+The **driver_config** section of a Modbus-TK device configuration file supports a variety of parameter definitions,
+but only **device_address** is required:
+
+    - **name** (Optional) - Name of the device. Defaults to "UNKNOWN".
+    - **device_type** (Optional) - Name of the device type. Defaults to "UNKNOWN".
+    - **device_address** (Required) - IP Address of the device.
+    - **port** (Optional) - Port the device is listening on. Defaults to 0 (no port). Use port 0 for RTU transport.
+    - **slave_id** (Optional) - Slave ID of the device. Defaults to 1. Use ID 0 for no slave.
+    - **baudrate** (Optional) - Serial (RTU) baud rate. Defaults to 9600.
+    - **bytesize** (Optional) - Serial (RTU) byte size: 5, 6, 7, or 8. Defaults to 8.
+    - **parity** (Optional) - Serial (RTU) parity: none, even, odd, mark, or space. Defaults to none.
+    - **stopbits** (Optional) - Serial (RTU) stop bits: 1, 1.5, or 2. Defaults to 1.
+    - **xonxoff** (Optional) - Serial (RTU) flow control: 0 or 1. Defaults to 0.
+    - **addressing** (Optional) - Data address table: offset, offset_plus, or address. Defaults to offset.
+        - address: The exact value of the address without any offset value.
+        - offset: The value of the address plus the offset value.
+        - offset_plus: The value of the address plus the offset value plus one.
+        - : If an offset value is to be added, it is determined based on a point's properties in the CSV file:
+            - Type=bool, Writable=TRUE:       0
+            - Type=bool, Writable=FALSE:  10000
+            - Type!=bool, Writable=TRUE:  30000
+            - Type!=bool, Writable=FALSE: 40000
+    - **endian** (Optional) - Byte order: big, little, or mixed. Defaults to big.
+    - **write_multiple_registers** (Optional) - Write multiple coils or registers at a time. Defaults to true.
+        - : If write_multiple_registers is set to false, only register types unsigned short (uint16) and boolean (bool)
+        is supported. The exception raised during the configure process.
+    - **selected_registers** (Optional) - Comma-separated list of "Register Name" values, selected from the
```

Comment:
This should probably be a list of strings. Then you can just use the VOLTTRON Point Names instead of a new redundant column.

Reply:
I'll change the selected_registers to the list of VOLTTRON point name

Reply:
Also, if you wanted to allow selecting nothing and everything you could use a null value for everything and an empty list for nothing.

Reply:
The selected_registers section is either select all registers or select some registers from the registry_config. Is there any case that you would select nothing?

Reply:
I don't think so but I'm usually inclined to include options that I don't see an immediate use for if they just fall out of a simple implementation.
I see no problem with leaving out a "select nothing" option if there is no use case for it.